postgres/pgdump.go

```diff
@@ -699,45 +707,54 @@ func getCols(conv *internal.Conv, table string, l []nodes.Node) (cols []string,
-func getVals(conv *internal.Conv, l [][]nodes.Node, n nodes.InsertStmt) (values []string) {
+func getVals(conv *internal.Conv, l []*pg_query.Node, n *pg_query.InsertStmt) (values []string) {
 	for _, vl := range l {
-		for _, v := range vl {
-			switch c := v.(type) {
-			case nodes.A_Const:
-				switch st := c.Val.(type) {
-				case nodes.String:
-					values = append(values, st.Str)
-				case nodes.Integer:
-					// For uniformity, convert to string and handle everything in
-					// dataConversion(). If performance of insert statements becomes a
-					// high priority (it isn't right now), then consider preserving int64
-					// here to avoid the int64 -> string -> int64 conversions.
-					values = append(values, strconv.FormatInt(st.Ival, 10))
+		switch vals := vl.GetNode().(type) {
+		case *pg_query.Node_List:
+			for _, val := range vals.List.Items {
+				switch c := val.GetNode().(type) {
+				case *pg_query.Node_AConst:
+					switch st := c.AConst.Val.GetNode().(type) {
+					case *pg_query.Node_String_:
+						values = append(values, trimString(st.String_))
+					case *pg_query.Node_Integer:
+						// For uniformity, convert to string and handle everything in
+						// dataConversion(). If performance of insert statements becomes a
+						// high priority (it isn't right now), then consider preserving int64
+						// here to avoid the int64 -> string -> int64 conversions.
+						values = append(values, strconv.FormatInt(int64(st.Integer.Ival), 10))
+					default:
+						conv.Unexpected(fmt.Sprintf("Processing %v statement: found %s node for A_Const Val", reflect.TypeOf(n), reflect.TypeOf(st)))
+					}
 				default:
-					conv.Unexpected(fmt.Sprintf("Processing %v statement: found %s node for A_Const Val", reflect.TypeOf(n), reflect.TypeOf(c.Val)))
+					conv.Unexpected(fmt.Sprintf("Processing %v statement: found %s node in ValuesList", reflect.TypeOf(n), reflect.TypeOf(c)))
 				}
-			default:
-				conv.Unexpected(fmt.Sprintf("Processing %v statement: found %s node in ValuesList", reflect.TypeOf(n), reflect.TypeOf(v)))
 			}
+		default:
+			conv.Unexpected(fmt.Sprintf("Processing %v statement: found %s in ValuesList", reflect.TypeOf(n), reflect.TypeOf(vals)))
 		}
 	}
 	return values
 }
 
-func logStmtError(conv *internal.Conv, n nodes.Node, err error) {
-	conv.Unexpected(fmt.Sprintf("Processing %v statement: %s", reflect.TypeOf(n), err))
-	conv.ErrorInStatement(prNodes([]nodes.Node{n}))
+func logStmtError(conv *internal.Conv, node interface{}, err error) {
+	conv.Unexpected(fmt.Sprintf("Processing %v statement: %s", reflect.TypeOf(node), err))
+	conv.ErrorInStatement(PrNodeType(node))
 }
 
-func getString(node nodes.Node) (string, error) {
-	switch n := node.(type) {
-	case nodes.String:
-		return n.Str, nil
+func getString(node *pg_query.Node) (string, error) {
+	switch n := node.GetNode().(type) {
+	case *pg_query.Node_String_:
+		return trimQuote(strings.TrimPrefix(n.String_.String(), "str:")), nil
```

Comment:
Is it a good idea to replace this with trimString()?

Reply:
Done @ https://github.com/cloudspannerecosystem/harbourbridge/pull/172/commits/3701d42b42a0409b2981d46d8efa2c4f2ef60ba4.